src/core/primitives/Primitive/index.tsx

```
@@ -27,28 +28,10 @@ const createPrimitiveComponent = (elementType: SupportedElement) => {
             const child = childrenArray[0] as React.ReactElement;
 
             // Merge refs if child already has one
-            // TODO: This can be made into a utility function
-            const mergedRef = (childRef: any) => {
-                if (typeof ref === 'function') ref(childRef);
-                else if (ref) ref.current = childRef;
+            // We prioritize the child's props over elementProps, but merge classNames
+            const finalProps = useMergeProps(elementProps, child, ref);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)